spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java

```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
 	}
+
+
+	@Test
+	public void testOne() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.putAll(basicMapId);
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+	}
+
+
+	@Test
+	public void testTwo() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		int intOne = basicMapId.size();
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+
+		assertEquals(0, intOne);
+	}
+
+
+	@Test
+	public void testThree() throws Exception {
+
+		BasicMapId basicMapId = (BasicMapId) BasicMapId.id("#xAD]-}", (Serializable) "#xAD]-}");
+
+		assertEquals(1, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		Serializable serializable = basicMapId.put("", (Serializable) "#xAD]-}");
+
+		assertEquals(2, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertNull(serializable);
+
+		String string = (String) basicMapId.put("", (Serializable) "#xAD]-}");
+
+		assertEquals(2, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertNotNull(string);
+		assertEquals("#xAD]-}", string);
+	}
+
+
+	@Test
+	public void testFour() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		HashMap<String, String> hashMap = new HashMap<String, String>();
+		Serializable serializable = basicMapId.put("", (Serializable) hashMap);
+
+		assertEquals(1, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertEquals(0, hashMap.size());
+		assertTrue(hashMap.isEmpty());
+
+		assertNull(serializable);
+
+		HashMap hashMapTwo = (HashMap) basicMapId.get("");
+
+		assertEquals(1, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertTrue(hashMapTwo.isEmpty());
+		assertEquals(0, hashMapTwo.size());
+	}
+
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testFiveRaisesIllegalArgumentException() throws Exception {
+
+		BasicMapId.id((MapId) null);
+	}
+
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testSevenRaisesIllegalArgumentException() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId((Map<String, Serializable>) null);
+	}
+
+
+	@Test
+	public void testEight() throws Exception {
+
+		BasicMapId basicMapId = (BasicMapId) BasicMapId.id("#xAD]-}", (Serializable) "#xAD]-}");
+
+		assertFalse(basicMapId.isEmpty());
+		assertEquals(1, basicMapId.size());
+
+		int intOne = basicMapId.size();
+
+		assertFalse(basicMapId.isEmpty());
+		assertEquals(1, basicMapId.size());
+
+		assertEquals(1, intOne);
+	}
+
+
+	@Test
+	public void testNine() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.hashCode();
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+	}
+
+
+	@Test
+	public void testTen() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		Set<Map.Entry<String, Serializable>> set = (Set<Map.Entry<String, Serializable>>) basicMapId.entrySet();
```

Comment:
It tests that an empty entrySet gets returned.
I've renamed the test in addition.